definitions.h

```diff
@@ -282,18 +267,238 @@ class Attributes
+            four[1][1] = 1;
+            four[2][2] = 1;
+            four[3][3] = 1;
+
+            //Single vertex
+            result.x = 0;
+            result.y = 0;
+            result.z = 0;
+            result.w = 0;
+
+            //Based on the first number in the uniforms
+            //the wanted transformation is performed
+            switch((int)uniforms.other[0])
+            {
+                case 0:
+                translate(vertIn, uniforms);
+                break;
+                case 1:
+                scale(vertIn, uniforms);
+                break;
+                case 2:
+                rotate(vertIn, uniforms);
+                break;
+                default:
+                STR(vertIn, uniforms);
+            }
+        }
+
+        //Translate the vertex by the uniform using matrix multiplication
+        void translate(const Vertex & vertIn, const Attributes & uniforms)
+        {
+            double sum[4] = {0, 0, 0, 0};
+            double vert[4] = {vertIn.x, vertIn.y, vertIn.z, vertIn.w};
+
+            //Setup the matrix
+            four[0][2] = uniforms.other[1];
+            four[1][2] = uniforms.other[2];
+
+            //Multiply and add the result
+            for(int i = 0; i < 4; i++)
+            {
+                for (int j = 0; j < 4; j++)
+                {
+                    sum[i] += four[i][j] * vert[j];
+                }
+            }
+
+            //Save the result
+            result.x = sum[0];
+            result.y = sum[1];
+            result.z = sum[2];
+            result.w = sum[3];
+        }
+
+        //Rotate the vertex by the uniform using matrix multiplication
+        void rotate(const Vertex & vertIn, const Attributes & uniforms)
+        {
+            //Convert degrees to radians
+            double angle = (uniforms.other[1] * 3.14) / 180;
+            double sum[4] = {0, 0, 0, 0};
+            double vert[4] = {vertIn.x, vertIn.y, vertIn.z, vertIn.w};
+
+            //Setup the matrix
+            four[0][0] = cos(angle);
+            four[0][1] = -sin(angle);
+            four[1][0] = sin(angle);
+            four[1][1] = cos(angle);
+
+            //Multiply and add the result
+            for(int i = 0; i < 4; i++)
+            {
+                for (int j = 0; j < 4; j++)
+                {
+                    sum[i] += four[i][j] * vert[j];
+                }
+            }
+
+            //Save the result
+            result.x = sum[0];
+            result.y = sum[1];
+            result.z = sum[2];
+            result.w = sum[3];
+        }
+
+        //Scale the vertex by the uniform using matrix multiplication
+        void scale(const Vertex & vertIn, const Attributes & uniforms)
+        {
+            double sum[4] = {0, 0, 0, 0};
+            double vert[4] = {vertIn.x, vertIn.y, vertIn.z, vertIn.w};
+
+            //Setup the matrix
+            four[0][0] = uniforms.other[1];
+            four[1][1] = uniforms.other[1];
+
```

Comment:
it would be cool to override the multiplication operator as we discussed in class,  to be more human readable